virtualizarr/manifests/manifest.py

```diff
@@ -35,16 +42,23 @@ def __repr__(self) -> str:
         return f"ChunkEntry(path='{self.path}', offset={self.offset}, length={self.length})"
 
     @classmethod
-    def from_kerchunk(cls, path_and_byte_range_info: list[str | int]) -> "ChunkEntry":
-        path, offset, length = path_and_byte_range_info
+    def from_kerchunk(
+        cls, path_and_byte_range_info: tuple[str] | tuple[str, int, int]
+    ) -> "ChunkEntry":
+        if len(path_and_byte_range_info) == 1:
```

Comment:
I think this could be an `isinstance(..., str)` check instead perhaps? Kerchunk's [docs on its specification](https://fsspec.github.io/kerchunk/spec.html) indicate that it should return either a bare `str` or a `tuple[str, int, int]`. We can't predict when kerchunk will randomly decide to return the former, but we should be as explicit as possible when checking to see if that is what has happened. (I'm worried that there may be other situations in which kerchunk returns a length-1 something that isn't actually a string...)

Reply:
Ooh, excellent catch! Kerchunk doesn't like to inline Zarr variable data, effectively masking the case you point out.

I just lazily turned it into a `NotImplementedError` for now since it would require a bit more thought. (Probably best to follow up in a subsequent PR.)

Reply:
Sorry, when does the inline case come up here exactly? I feel like it is also relevant for #119?

Reply:
Here's a simple reproducer:

```python
import kerchunk.zarr
import xarray as xr

ds = xr.Dataset({}, coords={"time": [0, 1, 2]})
ds.to_zarr("/tmp/ds.zarr", mode="w")
ref = kerchunk.zarr.ZarrToZarr("/tmp/ds.zarr").translate()
ref["refs"]["time/0"]  # ['file:///tmp/ds.zarr/time/0']
```

I think it's a kerchunk bug that Zarrs are never inlined, even when `inline_threshold` is set.

Reply:
> I think it's a kerchunk bug that Zarrs are never inlined, even when inline_threshold is set.

That doesn't surprise me - we already had to deal with https://github.com/fsspec/kerchunk/issues/445.

But we actually don't use kerchunk's inlining feature at all in virtualizarr - instead if we want eager access to some variable we just load it as a numpy array-backed xarray variable instead (
https://virtualizarr.readthedocs.io/en/latest/usage.html#loading-variables). So we just set `inline_threshold=0` whenever we call any kerchunk reader internally.

But I'm a little confused - that's all separate from how kerchunk will sometimes return a `str` instead of `tuple[str, int, int]`. Your example `ref["refs"]["time/0"]  # ['file:///tmp/ds.zarr/time/0']` is not an example of inlining, it's just kerchunk using a different way to implicitly represent the byte range reference to an entire chunk.

Also as I said above I don't think we should call `kerchunk.zarr.ZarrToZarr` inside `virtualizarr.open_virtual_dataset`, we should rewrite that function ourselves (but not in this PR).

Reply:
Oh, sorry, I gave you an example for the wrong case.

I don't actually have any particular case in mind where data is inlined. I was assuming that inlined data is possible because kerchunk does it. Also I was hoping that I'd be able to use VirtualiZarr to open references that I've already kerchunked, in which case it would certainly occur.

If the only intended use of kerchunk is internal, should I remove that check? Or at least reword the text so that it's not a TODO?

Reply:
> Also I was hoping that I'd be able to use VirtualiZarr to open references that I've already kerchunked, in which case it would certainly occur.

Currently the only intended use case of kerchunk is internal, but this case:

> Also I was hoping that I'd be able to use VirtualiZarr to open references that I've already kerchunked, in which case it would certainly occur.

we would like to support, which is being (intermittently) worked on in #119 and recently #186.

I think that the minimal version of this PR does not need to handle kerchunk-inlined data though, and we should merge this useful PR before worrying about that.